### ISSUE TITLE
False positives with NamespaceSniff 

### ISSUE THREAD

**Issue by schmengler** [magento-ecg/coding-standard#46]
In [#13e5071](https://github.com/magento-ecg/coding-standard/commit/13e5071d247097f2cd1490eb09fabc39d73fceb8) the "NamespaceSniff" sniffer was introduced, with the comment "When catching an exception inside a namespace it is important that you escape to the global space."

I assume that this is meant to prevent accidental `catch (Exception $e)` statements where `Exception` is not imported.

However it is also triggered by code like this:

```
namespace N1;

class CustomException extends \Exception {}

try {
    throw CustomException;
} catch (CustomException) {
}
```

which is valid good code. Using the FQN instead even triggers a warning by [PHP Inspections](https://plugins.jetbrains.com/plugin/7622-php-inspections-ea-extended-) in PhpStorm.

If you do not want to remove this check (which I would prefer), I suggest to limit it to "Exception" and "Mage_*Exception" classes. But note that this would still give a false positive with `N1\Exception`.

But it should definitely be a warning, and not an error.

**Comment by zhooravlik** [comment]
Hi @schmengler ,
Thanks for noticing possible issues, and thanks for the long-awaiting response :) 
The issue here - is not properly described purpose of the NamespaceSniff,- that's needed only for the non-imported Classes, see an example:
![image](https://user-images.githubusercontent.com/1256891/150536586-888c6659-0894-432c-9fc3-f88314be2a01.png)
Error is the appropriate type of sniffer results for the case.

For all other cases, this just will work (e.g. if you're using full class path like \N1\Exception or \RuntimeException or \Exception).
Closing the issue.
